src/main/java/ome/services/blitz/repo/ManagedRepositoryI.java

```diff
@@ -1286,6 +1383,60 @@ protected FsFile createTemplatePath(Object consistentData, Ice.Current __current
         return wholeExpanded;
 }
 
+    /**
+     * Expand the template path and create its directories with the correct ownership.
+     * Performs the work of {@link #createTemplatePath(Object, Current)} except from within an existing transaction.
+     * @param consistentData the object to hash in expanding {@code %hash%}
+     * @param ctx the event context to apply in expanding terms
+     * @param sf the service factory
+     * @param session the Hibernate session
+     * @param sql the JDBC convenience wrapper
+     * @return the expanded template path
+     * @throws ServerError if the new path could not be created
+     */
+    public FsFile createTemplatePath(Object consistentData, EventContext ctx, ServiceFactory sf, Session session, SqlAction sql)
```

Comment:
This method seems to do a surprising amount of heavy lifting.

Reply:
It does. It's basically a from-inside-executor translation of the existing one. Though, looking at it, I've not seen clear low-hanging fruit for refactoring.